lib/features/bendpoints/BendpointMove.js

```diff
@@ -151,4 +209,34 @@ BendpointMove.$inject = [
   'dragging',
   'rules',
   'modeling'
-];
\ No newline at end of file
+];
+
+
+// helpers //////////
+
+function isReverse(context) {
+  var hover = context.hover,
+      source = context.source,
+      target = context.target,
+      type = context.type;
+
+  if (type === 'reconnectStart') {
+    return hover && target && hover === target;
+  }
+
+  if (type === 'reconnectEnd') {
+    return hover && source && hover === source;
+  }
+}
+
+function replaceFirst(array, item) {
+  array.splice(0, 1, item);
+
+  return array;
+}
+
+function replaceLast(array, item) {
```

Comment:
This [replaces in place](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice), leaking a side effect so to say.

Reply:
This was removed later.